### PR TITLE
tls: TLS between TiDB components

### PR DESCRIPTION
**Pull request by weekface**
### What problem does this PR solve? 

This PR adds the new API to [Enable TLS Authentication](https://pingcap.com/docs/stable/how-to/secure/enable-tls-between-components/).

Part of #1855 

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has Go code change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Enable TLS between TiDB components
```

### CODE REVIEW

**Comment by Yisaer** [comment]
@weekface  Is this request ready to be reviewed or still WIP?

**Comment by weekface** [comment]
It is still WIP. Reviewers removed.

File: charts/tidb-cluster/templates/config/_prometheus-config.tpl [hunk 1]
```diff
@@ -21,10 +21,10 @@ scrape_configs:
   {{- end }}
   tls_config:
     insecure_skip_verify: true
-  {{- if .Values.enableTLSCluster }}
-    ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
-    cert_file: /var/lib/pd-client-tls/tls.crt
-    key_file: /var/lib/pd-client-tls/tls.key
+  {{- if and .Values.tlsCluster .Values.tlsCluster.enabled }}
+    ca_file: /var/lib/cluster-client-tls/ca.crt
```
Comment (weekface) [inline charts/tidb-cluster/templates/config/_prometheus-config.tpl]: all pods's client certificate mounted to `/var/lib/cluster-client-tls`.

File: pkg/manager/member/pd_member_manager.go [hunk 2]
```diff
@@ -206,16 +206,6 @@ func (pmm *pdMemberManager) syncPDStatefulSetForTidbCluster(tc *v1alpha1.TidbClu
 		if err != nil {
 			return err
 		}
-		if tc.IsTLSClusterEnabled() {
```
Comment (weekface) [inline pkg/manager/member/pd_member_manager.go]: removed certificates that issued by K8s built-in certificate system.

**Comment by cofyc** [comment]
https://github.com/pingcap/tidb-operator/blob/9d49e0448765da2b93f285a631b649ea42d9eead/tests/e2e/tidbcluster/tidbcluster.go#L145-L151

this case does not enable tls anymore, remove it and add a new test for tls between TiDB components?

**Comment by weekface** [comment]
> https://github.com/pingcap/tidb-operator/blob/9d49e0448765da2b93f285a631b649ea42d9eead/tests/e2e/tidbcluster/tidbcluster.go#L145-L151
> 
> this case does not enable tls anymore, remove it and add a new test for tls between TiDB components?

🙆‍♂ good catch.

**Comment by weekface** [comment]
@cofyc It is a little time consuming to add E2E test case for Cluster TLS. I marked it as a `TODO` in #1855, and i will do it in the future.

**Pull request merged**